### PR TITLE
add fmemopen, open_memstream, and open_wmemstream for POSIX.1-2008

### PR DESCRIPTION
**Pull request by jedbrown**
This is useful for interfacing with C libraries that can write output to `FILE` objects, for example.

### CODE REVIEW

**Comment by JohnTitor** [comment]
CI says:
```
  cargo:warning=/checkout/target/aarch64-linux-android/debug/build/libc-test-a9ad6fe1596a08f5/out/main.c:4439:24: error: use of undeclared identifier 'fmemopen'; did you mean 'freopen'?
  cargo:warning=                return fmemopen;
  cargo:warning=                       ^~~~~~~~
  cargo:warning=                       freopen
  cargo:warning=/android/ndk-aarch64/bin/../sysroot/usr/include/stdio.h:286:7: note: 'freopen' declared here
  cargo:warning=FILE* freopen(const char* __path, const char* __mode, FILE* __fp);
  cargo:warning=      ^
  cargo:warning=/checkout/target/aarch64-linux-android/debug/build/libc-test-a9ad6fe1596a08f5/out/main.c:4439:24: error: incompatible pointer types returning 'FILE *(const char *, const char *, FILE *)' (aka 'struct __sFILE *(const char *, const char *, struct __sFILE *)') from a function with result type 'FILE *(*)(void *, size_t, const char *)' (aka 'struct __sFILE *(*)(void *, unsigned long, const char *)') [-Werror,-Wincompatible-pointer-types]
  cargo:warning=                return fmemopen;
  cargo:warning=                       ^~~~~~~~
  cargo:warning=/checkout/target/aarch64-linux-android/debug/build/libc-test-a9ad6fe1596a08f5/out/main.c:4444:24: error: use of undeclared identifier 'open_memstream'
  cargo:warning=                return open_memstream;
  cargo:warning=                       ^
  cargo:warning=/checkout/target/aarch64-linux-android/debug/build/libc-test-a9ad6fe1596a08f5/out/main.c:4449:24: error: use of undeclared identifier 'open_wmemstream'
  cargo:warning=                return open_wmemstream;
  cargo:warning=                       ^
  cargo:warning=4 errors generated.
  exit code: 1
```

but I think the aarch64 env already has them, doesn't it?

**Comment by jedbrown** [comment]
I tried to model off other POSIX.1-2008 functions and I don't understand why it would be missing only on aarch64. What's the procedure for fixing this?

**Comment by JohnTitor** [comment]
> 
> 
> I tried to model off other POSIX.1-2008 functions and I don't understand why it would be missing only on aarch64. What's the procedure for fixing this?

If it isn't missing, we can skip the tests for them in `libc-test/build.rs`.

**Comment by JohnTitor** [comment]
And I think it's related to #1765, it'd be great if you also could add a FIXME comment(s) with this issue number.

**Comment by JohnTitor** [comment]
@jedbrown Friendly-ping, we're going to a new release. If you're still interested, I'm happy to wait for a while until this gets merged :)

**Comment by jedbrown** [comment]
@JohnTitor Thanks for the ping. I've added it to the skip list for aarch64. Did you also want a comment in `src/unix/mod.rs` next to these functions? I don't see it for the other cases listed in `libc-test/build.rs`.

**Review by JohnTitor — APPROVED** [review]
> Did you also want a comment in src/unix/mod.rs next to these functions? I don't see it for the other cases listed in libc-test/build.rs.

No, this makes sense now. Thanks!

**Pull request merged**
